### PR TITLE
Support to get engine id, name, url in KyuubiConnection

### PR DESCRIPTION
**Pull request by turboFei**
### _Why are the changes needed?_

Support to get engine id, name, url in KyuubiConnection.

It is helpful and friendly for BI.

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by turboFei** [comment]
cc @yaooqinn @wForget

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/KyuubiSparkUtil.scala [hunk 1]
```diff
@@ -62,9 +66,9 @@ object KyuubiSparkUtil extends Logging {
     // scalastyle:off line.size.limit
     // format: off
     s"""
-       |           Spark application name: ${sc.appName}
+       |           Spark application name: $engineName
        |                 application ID: $engineId
-       |                 application web UI: $webUrl
```
Comment (wForget) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/KyuubiSparkUtil.scala:67]: The webUrl(L63) is not used and needs to be removed.

**Review by wForget — COMMENTED** [review]
LGTM,

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/KyuubiSparkUtil.scala (hunk 1 above)
Comment (turboFei) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/KyuubiSparkUtil.scala:67]: resolved, thanks

**Review by wForget — APPROVED** [review]

File: kyuubi-server/src/main/scala/org/apache/kyuubi/client/KyuubiSyncThriftClient.scala [hunk 2]
```diff
@@ -164,8 +169,14 @@ class KyuubiSyncThriftClient private (
...
-      .filter(_.containsKey("kyuubi.engine.id"))
-      .map(_.get("kyuubi.engine.id"))
+      .filter(_.containsKey(KYUUBI_ENGINE_ID))
+      .map(_.get(KYUUBI_ENGINE_ID))
+    _engineName = Option(resp.getConfiguration)
+      .filter(_.containsKey(KYUUBI_ENGINE_NAME))
+      .map(_.get(KYUUBI_ENGINE_NAME))
+    _engineUrl = Option(resp.getConfiguration)
+      .filter(_.containsKey(KYUUBI_ENGINE_URL))
+      .map(_.get(KYUUBI_ENGINE_URL))
```
Comment (cxzl25) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/client/KyuubiSyncThriftClient.scala:179]: Here filter and map seem a bit redundant, get should be enough.

**Review by cxzl25 — APPROVED** [review]

**Comment by turboFei** [comment]
thanks @wForget @cxzl25 for the review, merging to master

File: kyuubi-server/src/main/scala/org/apache/kyuubi/client/KyuubiSyncThriftClient.scala (hunk 2 above)
Comment (cxzl25) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/client/KyuubiSyncThriftClient.scala:179]: Sorry, I thought it was scala map.

```scala
scala> Option(Map("key"->"v")).map(_.get("k")).flatten
res0: Option[String] = None
```

**Comment by turboFei** [comment]
I found that, if the user use the high version kyuubi jdbc including this pr to access low version kyuubi server.

It will throw `out of sequence` thrift exception.

**Comment by pan3793** [comment]
> I found that, if the user use the high version kyuubi jdbc including this pr to access low version kyuubi server.
> 
> It will throw `out of sequence` thrift exception.

it's critical, we need to fix it

**Pull request closed without merging**